packages/frontend/src/pages/Dashboard.tsx

```diff
@@ -235,14 +233,18 @@ const Dashboard: FC = () => {
                         `Error parsing chart in dashboard. Attempted to parse: ${unsavedDashboardTilesRaw} `,
                     );
                 }
+            } else {
+                // If there are no dashboard tiles, set them to the saved ones
+                // This is the first time the dashboard is being loaded.
+                if (!dashboardTiles) {
+                    setDashboardTiles(savedTiles);
+                }
             }
-
-            setDashboardTiles(unsavedDashboardTiles || savedTiles);
```

Comment:
This was the main problem. Sometimes this useEffect would run a second time after setting unsavedDashboardTiles, so it was undefined here and it set them back to the savedOnes.

Reply:
Great catch!! Multiple effects interacting, definitely not ideal